src/cli/commands/test/iac-local-execution/parsers/k8s-or-cloudformation-parser.ts

```
@@ -15,79 +8,41 @@ export function detectConfigType(
   fileData: IacFileData,
   parsedIacFiles: any[],
 ): IacFileParsed[] {
-  return parsedIacFiles.map((parsedIaCFile, docId) => {
-    if (
-      checkRequiredFieldsMatch(parsedIaCFile, REQUIRED_CLOUDFORMATION_FIELDS)
-    ) {
-      return {
-        ...fileData,
-        jsonContent: parsedIaCFile,
-        projectType: IacProjectType.CLOUDFORMATION,
-        engineType: EngineType.CloudFormation,
-        docId,
-      };
-    } else if (checkRequiredFieldsMatch(parsedIaCFile, REQUIRED_K8S_FIELDS)) {
-      return {
-        ...fileData,
-        jsonContent: parsedIaCFile,
-        projectType: IacProjectType.K8S,
-        engineType: EngineType.Kubernetes,
-        docId,
-      };
-    } else {
-      if (fileData.fileType === 'json') {
-        throw new FailedToDetectJsonConfigError(fileData.filePath);
+  return parsedIacFiles
+    .map((parsedFile, docId): IacFileParsed | null => {
+      if (
+        checkRequiredFieldsMatch(parsedFile, REQUIRED_CLOUDFORMATION_FIELDS)
+      ) {
+        return {
+          ...fileData,
+          jsonContent: parsedFile,
+          projectType: IacProjectType.CLOUDFORMATION,
+          engineType: EngineType.CloudFormation,
+          docId,
+        };
+      } else if (checkRequiredFieldsMatch(parsedFile, REQUIRED_K8S_FIELDS)) {
+        return {
+          ...fileData,
+          jsonContent: parsedFile,
+          projectType: IacProjectType.K8S,
+          engineType: EngineType.Kubernetes,
+          docId,
+        };
       } else {
-        throw new FailedToDetectYamlConfigError(fileData.filePath);
+        return null;
```

I'm wondering if this is the correct layer to swallow this error. We currently have `IacFileParsed` and `IacFileParseFailure`, I'm wondering if here we should be returning an `IacFileParseFailure` or introducing a third type e.g. `IacFileParseSkipped`.

Then we would filter these out at the result rendering layer by default. This would also allow us to present the total number of skipped files to the user as part of the output and in future we could potentially introduce a `--strict` flag that shows the errors, as I think there is some value in the error messages provided here if you're unclear on why your file was skipped.

Thoughts?

---

I think we have a bunch of options here. I'll try to lay out my mental model of it:

1. Provide no way for the user to see what's skipped
   * This is what the PR does at the time of writing, and is what is motivating your comment. I can see why this might give us pause, in case users are confused.
1. Print a list of all files skipped
   * Optionally behind a flag, e.g. `--show-skipped`?
   * You mention possibly introducing a `--strict` that treats unrecognised yaml/json as errors, but personally I don't agree with this... it's semantically never an error to skip an unsupported schema. IMO a possible problem with this PR currently is that it's hard for users to figure out _why_ a file they did expect to be scanned is skipped, in case we have a bug in our detection logic.
   * We currently print failure summaries in bold red. Should this list be above or below the failure summary? Maybe non-bold blue?
   * I worry this could be very noisy. An IaC-oriented repository might contain 1000s of json and yaml files.
   * Arguably non json/yaml/tf files are also "skipped" by IaC, and no-one is suggesting erroring/listing these!
1. Print a count of skipped files
   * Possibly behind a feature flag
   * Less noisy than printing them all, but there is information loss here. It doesn't really help the user debug what is going on.

---

@aron and I had a zoom about this, summary:

- skipped file diagnostics are probably more noise than signal, so we won't implement them here
- we should ensure we return an error to the user when no recognised types are found, as this is probably a mistake
   - when we find no valid extensions, we already return an error for this reason
   - implicitly, we had an error when no recognised IaC type was found because we errored on every unrecognised yaml/json 😅.